### PR TITLE
Update Go signing API

### PR DESCRIPTION
**Pull request by nick-drozd**

### CODE REVIEW

File: sdk/go/src/sawtooth_sdk/signing/core.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		return NewSecp256k1Context()
+	}
+
+	panic("No such algorithm")
+}
+
+// -- Signer --
+
+// A convenient wrapper of Context and PrivateKey.
+type signer struct {
```
Comment (aludvik) [inline sdk/go/src/sawtooth_sdk/signing/core.go]: Is this supposed to be private?

File: sdk/go/src/sawtooth_sdk/signing/core.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+// Returns the public key for this Signer instance.
+func (self *signer) GetPublicKey() PublicKey {
+	return self.context.GetPublicKey(self.private_key)
+}
+
+// -- CryptoFactory --
+
+// A factory for generating Signers.
+type cryptoFactory struct {
```
Comment (aludvik) [inline sdk/go/src/sawtooth_sdk/signing/core.go]: Is this supposed to be private? Making it private prevents the type from being used in any user-defined function.

Comment (nick-drozd) [reply sdk/go/src/sawtooth_sdk/signing/core.go]: I'll change those to public.

File: sdk/go/src/sawtooth_sdk/signing/core.go (hunk 1 above)
Comment (peterschwarz) [reply sdk/go/src/sawtooth_sdk/signing/core.go]: It should be public

**Pull request merged**